src/haz3lcore/dynamics/TypeAssignment.re

```diff
@@ -79,21 +80,23 @@ let dhpat_extend_ctx = (dhpat: DHPat.t, ty: Typ.t, ctx: Ctx.t): option(Ctx.t) =>
       // TODO: make this stricter
       let* ctrs = Typ.get_sum_constructors(ctx, ty);
       let* typ = ConstructorMap.get_entry(name, ctrs);
-      let* (ty1, ty2) = Typ.matched_arrow_strict(ctx, typ);
-      Typ.eq(ty2, ty) ? dhpat_var_entry(dhp, ty1) : None;
+      switch (typ) {
```

Comment:
why does this no longer do a matched arrow? also could be a let*

Reply:
I changed `ConstructorMap.get_entry` to return an `option(option(Typ.t))` rather than an arrow type for constructors due to some things elsewhere